pkg/start/start.go

```diff
@@ -195,6 +191,18 @@ func (o *Options) run(ctx context.Context, controllerCtx *Context, lock *resourc
 	controllerCtx.OpenshiftConfigManagedInformerFactory.Start(informersDone)
 	controllerCtx.InformerFactory.Start(informersDone)
 
+	allSynced := controllerCtx.CVInformerFactory.WaitForCacheSync(informersDone)
+	for _, synced := range allSynced {
+		if !synced {
+			klog.Fatalf("Caches never synchronized: %v", postMainContext.Err())
+			return
+		}
+	}
+	if err := controllerCtx.CVO.InitializeFromPayload(ctx, restConfig, burstRestConfig); err != nil {
```

Comment:
I'd argued earlier for this position, to get this disk-load out of the way before the possibly-slow wait for a lease.  But I'm going to reverse myself now, because I want #746 to go `ClusterOperatorDown{name="version"}` if we're waiting for a ClusterVersion for too long, and, since #528, we don't start  serving metrics until we have the leader lock.  Can we move this call to [right after the metrics launch][1]?  Sorry for waffling!

[1]: https://github.com/openshift/cluster-version-operator/blob/0e9bc4ef03004fcc2bb0c58d39e5b49445a9f8f6/pkg/start/start.go#L217

Reply:
Moved it. Still simply calling `Fatalf` on error to exit.